### PR TITLE
fix: stacking fee adjustment

### PR DESCRIPTION
**Pull request by kyranjamie**
> [Download the latest build](https://github.com/blockstack/stacks-wallet/actions/runs/512686849)

Now only adjusts stacking amount with the stacking libraries `modifyLockTxFee()` method, when the amount stacked is equal to, or within 1000 usTX of the address' balance.

Along with tx modal, stacking modal needs to be refactored.

### CODE REVIEW

File: app/modals/stacking/stacking-modal.tsx [hunk 1]
```diff
@@ -93,6 +93,11 @@ export const StackingModal: FC<StackingModalProps> = props => {
     balance: selectAddressBalance(state),
   }));
 
+  const accountBalance = new BigNumber(balance?.balance ?? 0);
+
+  const shouldModifyTxFee =
```
Comment (zone117x) [inline app/modals/stacking/stacking-modal.tsx:98]: I think I missed some context here, why 
> or within 1000 usTX of the address' balance. 
?

Comment (kyranjamie) [reply app/modals/stacking/stacking-modal.tsx:98]: You can't stack 100% of your balance as you need to pay the fee.

Comment (hstove) [reply app/modals/stacking/stacking-modal.tsx:98]: The fee rate in stacks-node is hard-coded per byte at the moment. So, until that changes, the fee will always be 260 ustx. Maybe we should just stick to that. If the fee rates change in stacks-node, then this adjustment might not even be enough, and we'll want to revisit this regardless of if 1000 is enough.

Comment (kyranjamie) [reply app/modals/stacking/stacking-modal.tsx:98]: Done

File: app/modals/stacking/stacking-modal.tsx [hunk 2]
```diff
@@ -93,6 +95,12 @@ export const StackingModal: FC<StackingModalProps> = props => {
...
+    accountBalance.isEqualTo(amountToStack) ||
+    accountBalance.minus(amountToStack).isLessThan(CONTRACT_CALL_FEE);
```
Comment (hstove) [inline app/modals/stacking/stacking-modal.tsx:102]: No need to have two checks here - if the first fails, the second will always fail. Maybe just use the second one.

**Review by hstove — APPROVED** [review]

File: app/modals/stacking/stacking-modal.tsx [hunk 3]
```diff
@@ -126,10 +132,12 @@ export const StackingModal: FC<StackingModalProps> = props => {
       burnBlockHeight: coreNodeInfo.burn_block_height,
     });
     const tx = await makeContractCall({ ...txOptions, senderKey: privateKey });
-    const modifiedFeeTx = stackingClient.modifyLockTxFee({
-      tx,
-      amountMicroStx: new BN(amountToStack.toString()),
-    });
+    const modifiedFeeTx = shouldModifyTxFee
```
Comment (zone117x) [inline app/modals/stacking/stacking-modal.tsx:135]: This `modifyLockTxFee` is pretty strange. Looks like it should be replaced with a function like `modifyStackStxTxAmount(amountToStack)`, and the tx fee should be modified with some generic function that works on any tx type.

**Review by zone117x — APPROVED** [review]

File: app/modals/stacking/stacking-modal.tsx (hunk 3 above)
Comment (kyranjamie) [reply app/modals/stacking/stacking-modal.tsx:135]: I agree, this method is quite confusing. This is fixed for now though, so will action.

**Comment by kyranjamie** [comment]
Merging without CI finish,  it'll run on the other branch anyway

**Pull request merged**